Composer/packages/client/src/recoilModel/dispatchers/auth.ts

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import formatMessage from 'format-message';
+import { CallbackInterface, useRecoilCallback } from 'recoil';
+import { AzureTenant } from '@botframework-composer/types';
+
+import {
+  userShouldProvideTokens,
+  isShowAuthDialog,
+  getTokenFromCache,
+  setTenantId,
+  getTenantIdFromCache,
+} from '../../utils/auth';
+import {
+  showAuthDialogState,
+  primaryTokenState,
+  graphTokenState,
+  availableTenantsState,
+  showTenantDialogState,
+  currentTenantState,
+} from '../atoms/appState';
+import { AuthClient } from '../../utils/authClient';
+import storage from '../../utils/storage';
+
+import { addNotificationInternal, createNotification } from './notification';
+
+export const authDispatcher = () => {
+  const setShowAuthDialog = useRecoilCallback((callbackHelpers: CallbackInterface) => (show: boolean) => {
```

Comment:
Tiny Nit: You can destructure the set here rather than importing CallbackInterface type.